worker/rhpv3.go

```diff
@@ -176,76 +178,88 @@ func (a *accounts) ResetDrift(ctx context.Context, id rhpv3.Account) error {
 }
 
 func (a *account) Convert() api.Account {
-	a.mu.RLock()
-	defer a.mu.RUnlock()
-	a.balanceMu.Lock()
-	defer a.balanceMu.Unlock()
+	a.rwmu.RLock()
+	defer a.rwmu.RUnlock()
+	a.mu.Lock()
+	defer a.mu.Unlock()
 	return api.Account{
-		ID:      a.id,
-		Balance: new(big.Int).Set(a.balance),
-		Drift:   new(big.Int).Set(a.drift),
-		Host:    a.host,
-		Owner:   a.owner,
+		ID:           a.id,
+		Balance:      new(big.Int).Set(a.balance),
+		Drift:        new(big.Int).Set(a.drift),
+		Host:         a.host,
+		Owner:        a.owner,
+		RequiresSync: a.requiresSync,
 	}
 }
 
 func (a *account) resetDrift(ctx context.Context) error {
-	a.mu.Lock()
-	defer a.mu.Unlock()
+	a.rwmu.Lock()
+	defer a.rwmu.Unlock()
 	if err := a.bus.ResetDrift(ctx, a.id); err != nil {
 		return err
 	}
-	a.balanceMu.Lock()
+	a.mu.Lock()
 	a.drift.SetInt64(0)
-	a.balanceMu.Unlock()
+	a.mu.Unlock()
 	return nil
 }
 
 // WithDeposit increases the balance of an account by the amount returned by
 // amtFn if amtFn doesn't return an error.
 func (a *account) WithDeposit(ctx context.Context, amtFn func() (types.Currency, error)) error {
-	a.mu.RLock()
-	defer a.mu.RUnlock()
+	a.rwmu.RLock()
+	defer a.rwmu.RUnlock()
 	amt, err := amtFn()
 	if err != nil {
 		return err
 	}
-	a.balanceMu.Lock()
+	a.mu.Lock()
 	a.balance = a.balance.Add(a.balance, amt.Big())
-	a.balanceMu.Unlock()
+	a.mu.Unlock()
 	return a.bus.AddBalance(ctx, a.id, a.owner, a.host, amt.Big())
 }
 
 // WithWithdrawal decreases the balance of an account by the amount returned by
 // amtFn if amtFn doesn't return an error.
 func (a *account) WithWithdrawal(ctx context.Context, amtFn func() (types.Currency, error)) error {
-	a.mu.RLock()
-	defer a.mu.RUnlock()
+	a.rwmu.RLock()
+	defer a.rwmu.RUnlock()
 	amt, err := amtFn()
+	if err != nil && strings.Contains(err.Error(), "ephemeral account balance was insufficient") {
+		a.mu.Lock()
+		a.requiresSync = true
+		a.mu.Unlock()
+		err2 := a.bus.SetRequiresSync(ctx, a.id, a.owner, a.host, true)
```

Comment:
`nit` in my head it was going to be called _schedule_ sync and have it be a timestamp so we have the added information of when a sync was scheduled and can tell whether something is going wrong with the syncing process, do you think that makes sense to refactor in?

Reply:
I'd honestly just log whenever we set the flag and reset it. Then when you see a failed sync in the logs you can just grep for when we set the flag the last time rather than curling an endpoint. So it's all nicely in one place.